golem-worker-service-base/src/gateway_execution/gateway_binding_executor.rs

```diff
@@ -138,50 +166,167 @@ impl<N> DefaultGatewayBindingExecutor<N> {
+                    Ok(worker_response) => {
+                        self.file_server_binding_handler
+                            .handle_file_server_binding_result(
+                                &resolved_binding.namespace,
+                                &resolved_binding.worker_detail,
+                                worker_response,
+                            )
+                            .await
+                            .to_response(&binding.request_details, session_store)
+                            .await
+                    }
                     Err(err) => {
-                        err.to_response(&binding.request_details, &resolved_binding.middlewares)
+                        err.to_response_from_safe_display(|_| StatusCode::INTERNAL_SERVER_ERROR)
                     }
                 }
             }
             Err(err_response) => err_response,
         }
     }
+
+    async fn handle_http_auth_call_binding<R>(
+        &self,
+        binding: &ResolvedGatewayBinding<N>,
+        security_scheme_with_metadata: &SecuritySchemeWithProviderMetadata,
+        session_store: &GatewaySessionStore,
+    ) -> R
+    where
+        AuthCallBackResult: ToResponse<R>,
+    {
+        match &binding.request_details {
+            GatewayRequestDetails::Http(http_request) => {
+                let authorisation_result = self
+                    .auth_call_back_binding_handler
+                    .handle_auth_call_back(
+                        http_request,
+                        security_scheme_with_metadata,
+                        session_store,
+                    )
+                    .await;
+
+                authorisation_result
+                    .to_response(&binding.request_details, session_store)
+                    .await
+            }
+        }
+    }
+
+    async fn redirect_or_continue<R>(
+        request_details: &GatewayRequestDetails,
+        session: &GatewaySessionStore,
+        middlewares: Middlewares,
+    ) -> Option<R>
+    where
+        HttpRequestAuthentication: MiddlewareIn<R>,
+        CorsPreflight: MiddlewareOut<R>,
+        MiddlewareInError: ToResponseFromSafeDisplay<R>,
+    {
+        let input_middleware_result = middlewares
+            .process_middleware_in(session, request_details)
+            .await;
+
+        match input_middleware_result {
+            Ok(incoming_middleware_result) => match incoming_middleware_result {
+                MiddlewareSuccess::Redirect(response) => Some(response),
+                MiddlewareSuccess::PassThrough => None,
+            },
+
+            Err(err) => {
+                Some(err.to_response_from_safe_display(|_| StatusCode::INTERNAL_SERVER_ERROR))
+            }
+        }
+    }
 }
 
 #[async_trait]
-impl<N: Send + Sync, R: Debug + Send + Sync> GatewayBindingExecutor<N, R>
+impl<N: Send + Sync + Clone, R: Debug + Send + Sync> GatewayBindingExecutor<N, R>
```

Comment:
Please don't use single letter type parameters - it is completely unreadable (for me). Let's stick to the verbose ones as before

Reply:
Done, fixed :+1
Thanks for spotting it